mitodl/mitxonline
# Updates `configure_for_dedp` command to make it more generic

#### Pre-Flight checklist


#### What are the relevant tickets?

none

#### What's this PR do?

Updates the `configure_for_dedp` command in the following ways:
- Renames it to `configure_tiers`
- Adds some options to specify the program's readable ID, title, and abbreviation (for use in the discount codes)
- Adds an option to specify a CSV file with the tier data in it

The defaults remain the same - without options, it'll set up tiers for DEDP - but specifying the options will allow you to generate tiers for other programs (and will create the program in the system if necessary). 

Additionally, it makes some minor styling changes so the text isn't red for everything. 

#### How should this be manually tested?

Automated tests should pass.

_To test defaults (i.e., DEDP):_ run `configure_tiers` with no options.

_To test changing program options:_ run `configure_tiers` with `--program <new readable ID>`, `--program-name "Program Name"`, and/or `--program-abbrev XXXX". The program should be created with the values specified. Omitting any of these will use the defaults:

| Option | Default |
|--------|---------|
| `--program` | `program-v1:MITx+DEDP` |
| `--program-name`  |  `Data, Economics and Development Policy` |
| `--program-abbrev`  | `DEDP` |

Note that the command will _update_ if it finds a matching readable ID for the program, so if you don't supply `--program` but supply either of the other two options, it will change your existing DEDP program record. 

_To test custom tiers:_ 

1. Create a CSV file with the following fields: `threshold,discount type,discount amount`
    Do not include a header. `discount type` should be one of `percent-off`, `dollars-off`, or `fixed-price`. A tier with threshold 0 must exist - the command will refuse to run if one isn't specified. Specifying tier data completely overrides the default tier settings, so you must provide all the tiers you wish to create.
2. Run the command with the `--tier-info <filename>` option. (You can combine this with program attributes if you want.) The command should create tiers as specified. 
 
As per usual, if the program/tier data exists, it should reuse existing discounts and tiers and all that. (It did this previously; it should continue to do that.)
